src/cocoa/toga_cocoa/widgets/detailedlist.py

```diff
@@ -0,0 +1,142 @@
+
+    @objc_method
+    def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
+        value = self.interface.data[row]
+        try:
+            data = value._impl
+        except AttributeError:
+            data = TogaData.alloc().init()
+            value._impl = data
+
+        data.attrs = {
+            attr: attr_impl(value, attr, self.interface.factory)
+            for attr in value._attrs
+        }
+
+        return data
+
+    # TableDelegate methods
+    @objc_method
+    def tableViewSelectionDidChange_(self, notification) -> None:
+        self.interface.selection = notification.object.selectedRow
+        self.interface.selected = self.interface.data[notification.object.selectedRow]
+        if self.interface.on_select:
+            row = notification.object.selectedRow if notification.object.selectedRow != -1 else None
+            self.interface.on_select(self.interface, row=row)
+
+    @objc_method
+    def tableView_heightOfRow_(self, table, row: int) -> float:
+        return 48.0
+
+
+class DetailedList(Widget):
+    def create(self):
+        # Create a List, and put it in a scroll view.
+        # The scroll view is the _impl, because it's the outer container.
+        self.native = RefreshableScrollView.alloc().init()
+        self.native.interface = self.interface
+        self.native.hasVerticalScroller = True
+        self.native.hasHorizontalScroller = False
+        self.native.autohidesScrollers = False
+        self.native.borderType = NSBezelBorder
+
+        # Create the List widget
+        self.detailedlist = TogaList.alloc().init()
+        self.detailedlist.interface = self.interface
+        self.detailedlist._impl = self
+        self.detailedlist.columnAutoresizingStyle = NSTableViewUniformColumnAutoresizingStyle
+
+        self.native.detailedlist = self.detailedlist
+
+        # Create the column for the detailed list
+        column = NSTableColumn.alloc().initWithIdentifier('data')
+        self.detailedlist.addTableColumn(column)
+        self.columns = [column]
+
+        cell = TogaDetailedCell.alloc().init()
+        column.dataCell = cell
+
+        cell.editable = False
+        cell.selectable = False
+
+        # Hide the column header.
+        self.detailedlist.headerView = None
+
+        self.detailedlist.delegate = self.detailedlist
+        self.detailedlist.dataSource = self.detailedlist
+
+        # Embed the tree view in the scroll view
+        self.native.documentView = self.detailedlist
+
+        # Add the layout constraints
+        self.add_constraints()
+
+    def change_source(self, source):
+        self.detailedlist.reloadData()
+
+    def insert(self, index, item):
+        self.detailedlist.reloadData()
+
+    def change(self, item):
+        self.detailedlist.reloadData()
+
+    def remove(self, item):
+        self.detailedlist.reloadData()
+
+    def clear(self):
+        self.detailedlist.reloadData()
```

Comment:
It "smells weird" that all of these methods are calling the same underlying method. Do we expect these to be overridden in use?

Reply:
No - it's an artefact of providing a single interface that provides enough detail for all backends to use.

Cocoa has a very purist "delegated data source" model for table/tree rendering. You don't add and remove elements to the visual tree/table; you specify a delegate that answers the question "what should I display at row X, column Y" on demand. `reloadData` is the way you tell the table "BTW - something visible has changed, please do a redraw". No matter the change, the visual response is "... and redraw".

GTK and Winforms, however, require a specific response to "add", "remove", "change", and so on, that insert, remove and modify specific nodes in the tree. So the interface needs to provide all the methods to support GTK and and Winforms; Cocoa (and FTR, iOS) just need to say "yup, got that; rerender".